util/fqdn.go

```diff
@@ -62,7 +62,7 @@ func GetAuthority(uri string) string {
 func GetHost(auth string) string {
 	begin := strings.Index(auth, "@")
 	if begin == -1 || begin == len(auth)-1 {
-		return ""
+		begin = -1
```

Comment:
I'm not sure if I understand the logic of setting `begin := -1` when `begin == -1`.